docs/04.guides/04.cookbooks/29.Threads-Scope/page.md

```diff
@@ -0,0 +1,272 @@
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	sleep(1000);
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+This below example shows thread execution. Code with in the thread tag run parallel to execution. Here execution doesn't wait for sleep.
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	thread {
+		sleep(1000);
+	}
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+Threads were independent form current running thread on lucee. Lucee doesn't wait for end of thread.
+
+Threads were mainly used to retrieve the data form DB, cfhttp, webservice it doesn't care about how much time it takes to execute.
+
+### Example 2 ###
+
+Here we can see about multiple threads in lucee. Lucee can also used to create multiple threads running parallel at that time.
+
+```lucee
+<cfscript>
+function napASecond(index) localmode=true {
+	thread  {
+		thread.start=now();
+		sleep(1000)
+		thread.end=now();
+	}
+}
+
+start=getTickCount();
+loop from=1 to=5 index="index" {
+	napASecond(index);
+}
+
+// show threads
+dump(var:cfthread,expand:false);
+// wait for all threads to finish
+thread action="join" name=cfthread.keyList();
+// show threads
+dump(var:cfthread,expand:false);
+
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+Above example shows currently five threads were running parallel, Thread action join wait for all threads to finish.
+
+```cfthread``` returns struct info of all thread status.
+
+
+Same example within tag
+
+```lucee
+<cffunction name="napASecond">
+	<cfargument name="index">
+	<cfthread index="#index#" action="run" name="t#index#">
+		<cfset thread.start=now()>
+		<cfset sleep(1000+(index*10))>
+		<cfset thread.end=now()>
+	</cfthread>	
+</cffunction>
+
+<cfset start=getTickCount()>
+
+<cfloop from="1" to="5" index="index">
+	<cfset napASecond(index)>
+</cfloop>
+
+<!-- show threads -->
+<cfdump var="#cfthread#" expand="false">
+
+<!-- wait for all threads to finish -->
+<cfthread action="join" name="t1,t2,t3">
+
+<!-- show threads -->
+<cfdump var="#cfthread#" expand="false">
+
+<cfdump var="done in #getTickCount()-start#ms">
+
+```
+
+Here it join three threads "t1,t2,t3". We can join threads by its name.
+
+
+### Example 3 ###
+
+Here we can see about threads running parallel using ```each``` function.
+
+```lucee
+<cfscript>
+tasks = ["Susi","Sorglos"];
+start=getTickCount();
+tasks.each(
+	function(value, index, array){
+		systemOutput(value,true);
+		sleep(1000);
+	}
+	,true
+);
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+```each``` function has a option parallel, when we set parallel attribute as true. It runs each element parallel. In default it should be false.
+
+So that two array elements runs parallel done in 1second.
+
```

Comment:
***ToChange***
***This is for lines 130 through 132***
The ```each``` function has an optional attribute called _parallel_. When we set the parallel attribute as true, it runs each element in parallel. By default, this attribute is false.
+
+Since the two array elements run in parallel, this code executes in 1 second.